### ISSUE TITLE
`-param type=label` cannot be used in `sct_register_to_template` because the `ilabel` and `dlabel` flags do not existe in `sct_register_to_template`

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#3600]
This registration option is possible with `sct_register_multimodal` but not with `sct_register_to_template` because the `ilabel` and `dlabel` flags do not existe in `sct_register_to_template`.

Solution: 
- introduce `ilabel` and `dlabel` flags

**Comment by joshuacwnewton** [comment]
@jcohenadad I looked into the script, and noticed that we actually do use the labels already for registration! 

But... Only for the `-ref subject` case:

   https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/7fd2ea718751dd858840c3823c0830a910d9777c/spinalcordtoolbox/scripts/sct_register_to_template.py#L745-L747
    
(Note the use of `fname_src_label` and `fname_dest_label`.) 

And, if you trace the code, `ftmp_label` comes from `fname_landmarks` (i.e. from the `-l`, `-ldisc`, or `-lspinal` arguments). So, we already have access to the necessary label files, without the need for additional `-ilabel` or `-dlabel` arguments.

----

By comparison, here is the `-ref template` case:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/7fd2ea718751dd858840c3823c0830a910d9777c/spinalcordtoolbox/scripts/sct_register_to_template.py#L685-L687

We don't pass `fname_src_label` or `fname_dest_label`  at all. So, it might just be a matter of updating this call to `register_wrapper()` to use the existing label files?

**Comment by jcohenadad** [comment]
> We don't pass `fname_src_label` or `fname_dest_label` at all.

Indeed, in the `-ref template` case, the label-based alignment is done earlier in the code:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/7fd2ea718751dd858840c3823c0830a910d9777c/spinalcordtoolbox/scripts/sct_register_to_template.py#L600

> So, it might just be a matter of updating this call to register_wrapper() to use the existing label files?

After thinking about it, I'm not sure we should introduce that feature (ie: the type=label option). In fact, the solution proposed in [this post](https://forum.spinalcordmri.org/t/labels-for-lumbar-spinal-level/737/8?u=jcohenadad) to create a 'fake' label 99 is, maybe, an acceptable workaround. In the future, we could consider making this label an official label. 

I cannot think of a scenario where users would like to have labels others than the ones that are already in the PAM50 template (except, of course, for the cauda equinea, and maybe other isolated cases). So maybe a resolution for this issue is to remove the 'type=label' option in the usage?
